pool.go

```diff
@@ -32,6 +33,8 @@ type Pool struct {
 	// exit is used to signal the pool to stop running the query loop
 	// and close the jobs channel
 	exit chan bool
+	// afterFunc is a function that gets called after each job.
+	afterFunc *func(*Job)
```

Comment:
Correct me if I'm wrong, but I don't think this needs to be a pointer to a function? Functions are, by themselves, already nillable. See http://play.golang.org/p/1qA9c70wpJ.